docs/rfcs/0001-designing-in-the-open.md

```diff
@@ -0,0 +1,148 @@
+participation. This document proposes how Grafana Agent will utilize public
+spaces for its primary home for future design proposals.
+
+## Goals
+
+* Outline options for proposing changes to Grafana Agent
+* Lower the barrier to entry for interested parties to become maintainers
+
+## Non-Goals
+
+* Enforce that every change originates from a fully public proposal or
+  discussion. While all maintainers and contributors will be encouraged to
+  design openly, there may be legal, security, privacy, or business reasons
+  that prevent some or all context from being made public.
+
+* Be overly prescriptive: too many rules can hinder adoption of a process. This
+  outlines more options than it is requirements.
+
+## Proposal
+
+Public proposals may take one of two forms:
+
+* Issue proposals
+* RFC PR proposals (e.g., this document)
+
+### Issues
+
+Issues are the quickest path towards proposing a change. Issue proposals must
+be opened at the [grafana/agent issues page](https://github.com/grafana/agent/issues).
+
+There are no strict set of rules for issue-based proposals, but authors are
+recommended to prefix the issue title with `Proposal:` so it may be found more
+easily.
+
+### RFC PRs
+
+RFC PR proposals must at least:
+
+* Be placed in the `docs/rfcs` folder of the `grafana/agent` repository
+* Have a lowercase filename in hyphen-case with an `.md` extension
+* Prefix the filename with the RFC ID
+  * ID `xxxx` may be initially used until the final ID is known
+* Contain valid markdown
+* Start with the title of the proposal
+* Contain a bullet point list of metadata of:
+  * The date the proposal was written
+  * The list of authors, with their names and GitHub usernames
+  * The PR where the proposal was posted
+
+`0000-template.md` contains a template to use for writing proposals that
+conforms to these rules.
+
+The remainder of the proposal may be formatted however the author wishes. Some
+example sections in the RFC may be:
+
+* Summary: What is the background that lead to this proposal?
+* Goals: What are the main goals of the proposal?
+* Non-Goals: What _aren't_ the main goals of the proposal?
+* Proposal: What is the proposal?
+* Pros: What are the upsides to this proposal?
+* Cons: What are the downsides to this proposal?
+* Considered Alternatives: Why is this proposal the best path forward? What were the alternatives?
+* Open Questions: What questions still need to be answered?
+* Prior Art: What was this proposal based on, if anything?
+
+#### RFC Review
+
+RFCs should be opened as a PR to grafana/agent, ideally prefixed in the PR
+title with `RFC:` to easily identify it amongst other PRs.
+
+#### RFC Mutability
+
+Once an RFC is accepted, it must be treated as immutable. There are two exceptions:
+
+* The author's name and username may be changed, as long as authorship is still
+  credited to the same individuals.
+
+* Errata and typos may be fixed as long as it does not change the
```

Comment:
Define how they are referenced, how deprecations, and referals to new RFCs work?

You don't _need_ to, but if they're "immutable", you need to be more specific about what's mutable, and how.

Reply:
👍 I've taken out the immutability requirement for now. Too much to think about for what we're trying to do at this stage.